### ISSUE TITLE
Using global settings as default instead of configuring them for every node separately

### ISSUE THREAD

**Issue by Gulpman** [timmbogner/Farm-Data-Relay-System#43]
For preventing configuration errors, better user experience as well as easier development, I think it would be a great decision to enable the use of the global config by default.
At the moment the default is:

```
//#include <fdrs_globals.h> //Uncomment when you install the globals file
[...]
#define BAND 915E6
#define SF 7
[...for gateways in addition...]
//WiFi and MQTT Credentials -- Needed only for MQTT gateway
#define WIFI_SSID   "Your SSID"  
#define WIFI_PASS   "Your Password"
#define MQTT_ADDR   "192.168.0.8"
#define MQTT_PORT   1883 // Default MQTT port is 1883

//MQTT Credentials -- Needed only if MQTT broker requires authentication
//#define MQTT_AUTH   //uncomment to enable MQTT authentication 
#define MQTT_USER   "Your MQTT Username"
#define MQTT_PASS   "Your MQTT Password"
```

This should be changed to:

```
#include <fdrs_globals.h> //Uncomment when you install the globals file
[...]
// LoRa settings - please set up in the global config. Only uncomment, if you need different settings for this sketch.
//#define BAND 915E6
//#define SF 7
[...for gateways in addition...]
//WiFi credentials - please set up in the global config. Only uncomment, if you need different settings for this sketch.
//#define WIFI_SSID   "Your SSID"  
//#define WIFI_PASS   "Your Password"
//MQTT Credentials - please set up in the global config. Only uncomment, if you need different settings for this sketch.
//#define MQTT_ADDR   "192.168.0.8"
//#define MQTT_PORT   1883 // Default MQTT port is 1883
//#define MQTT_AUTH   //uncomment to enable MQTT authentication 
//#define MQTT_USER   "Your MQTT Username"
//#define MQTT_PASS   "Your MQTT Password"
```

Pros: 
- preventing configuration errors: as settings only need to be defined once, there is only one chance to set the wrong value.
- better user experience: as a result the user runs into less issues, has to ask less questions and the system is up quicker
- easier development: as each developer has different settings, chances are less likely that developer-specific changes are commited to the source.

Cons: 
 - None. ;) Seriously I cannot find a con. Feel free to add :) 

This would lead to an easier onboarding for new users and does not take away any power for those who want to configure each node separately.

Code-wise the change is a no-brainer. It also would help to identify the pieces of code, where the implementation to check for global settings may be erronous. ;)

What do you think?

TL;DR: enable global settings by default, to prevent configuration errors and enhance first users experience.

**Comment by timmbogner** [comment]
In the coming three days I'm going to take the advice of @Devilbinder and move all common files to the root directory, so the user will need to drop the whole project directory into the libraries folder. This will eliminate the install folder and give the user little choice but to use global settings. Types will also go to the root. I hope this will release us from the headache of a gazillion identical files throughout the repo.

[The repo is having another uptick in traffic this week thanks to a fantastic article from HACKADAY](
https://hackaday.com/2022/07/02/farm-data-relay-system/)

**Comment by Gulpman** [comment]
> In the coming three days I'm going to take the advice of @Devilbinder and move all common files to the root directory, so the user will need to drop the whole project directory into the libraries folder. This will eliminate the install folder and give the user little choice but to use global settings. 

I'm looking forwar for the switch to a library (removing redundant files) and to use the global settings as default and leave individual settings for each sensor as an option. This will definitively make things easier for anyone - especially new users. 👍

>Types will also go to the root. I hope this will release us from the headache of a gazillion identical files throughout the repo.
>
Check my two pull requests: I exactly did this there, this way you can save yourself the work. :-)

> [The repo is having another uptick in traffic this week thanks to a fantastic article from HACKADAY](https://hackaday.com/2022/07/02/farm-data-relay-system/)

Awsome!

**Comment by timmbogner** [comment]
Most of the changes are now made to the dev branch. I hope things will be easier to keep organized now! Let me know if anything looks off.
